packages/components/src/draggable/index.js

```diff
@@ -19,19 +42,20 @@ export default function Draggable( {
 	transferData,
 	__experimentalDragComponent: dragComponent,
 } ) {
-	const dragComponentRef = useRef();
+	/** @type {import('react').MutableRefObject<HTMLDivElement | null>} */
+	const dragComponentRef = useRef( null );
 	const cleanup = useRef( () => {} );
 
 	/**
 	 * Removes the element clone, resets cursor, and removes drag listener.
 	 *
-	 * @param {Object} event The non-custom DragEvent.
+	 * @param {import('react').DragEvent} event The non-custom DragEvent.
 	 */
 	function end( event ) {
 		event.preventDefault();
 		cleanup.current();
 
-		if ( onDragOver ) {
+		if ( onDragEnd ) {
```

Comment:
That seems like an existing bug. Wondering how/what it does.. Looking at: https://github.com/WordPress/gutenberg/pull/26897/files

Reply:
I'm also not sure about this one. Any suggestions on how to test its impact?

Reply:
Sorry, I should have mentioned this in the PR description, I totally glossed over that this was a fix.

Not sure how to measure its impact, but theoretically any `onDragEnd` event that was supposed to be fired was not being fired due to the lack of an `onDragOver`. That sounds pretty bad in the context of this component 😬

Reply:
Maybe if we ping the author of the original code?

Reply:
@ellatrix As you did the original refactor to a function component, do you mind chiming in on this? Is there a good way to test the result of this bug fix?

Reply:
It is used: 

https://github.com/WordPress/gutenberg/blob/da8555d258edbe676fa079fb51252f918ae032e1/packages/block-editor/src/components/block-draggable/index.js#L76

https://github.com/WordPress/gutenberg/blob/da8555d258edbe676fa079fb51252f918ae032e1/packages/block-editor/src/components/inserter-draggable-blocks/index.js#L17

Drag and drop is hard to test, but manually by testing the block drag handle.

Reply:
Ah, I don't know how I missed those usages!

Luckily, this bug would never have affected those two usages. One does not use the `onDrag*` events passed to `Draggable` at all and the other passes both `onDragOver` and `onDragEnd` meaning that this bug wouldn't have affected it.

Block dragging continues to work as expected.

Reply:
That's good. No tests required imo.